sycl/doc/EnvironmentVariables.md

```diff
@@ -249,6 +249,7 @@ variables in production code.</span>
 | `SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE_FOR_FILL` | Integer | When set to a positive value enables use of a copy engine for memory fill operations. Default is 0. |
 | `SYCL_PI_LEVEL_ZERO_SINGLE_ROOT_DEVICE_BUFFER_MIGRATION` | Integer | When set to "0" tells to use single root-device allocation for all devices in a context where all devices have same root. Otherwise performs regular buffer migration. Default is 1. |
 | `SYCL_PI_LEVEL_ZERO_REUSE_DISCARDED_EVENTS` | Integer |  When set to a positive value enables the mode when discarded Level Zero events are reset and reused in scope of the same in-order queue based on the dependency chain between commands. Default is 1. |
+| `SYCL_PI_LEVEL_ZERO_EXPOSE_CSLICE_IN_AFFINITY_PARTITIONING` (Deprecated) | Integer | When set to non-zero value exposes compute slices as sub-sub-devices in `sycl::info::partition_property::partition_by_affinity_domain` partitioning scheme. Default is zero meaning that they are only exposed when partitioning by `sycl::info::partition_property::ext_intel_partition_by_cslice`. If the device doesn't support partitioning by compute slice this variable has no effect. This option is introduced for compatibility reasons and is immediately deprecated. New code must not rely on this behavior. Also note that even if sub-sub-device was created using `partition_by_affinity_domain` it would still be reported as created via partitioning by compute slices. |
```

Comment:
Are you adding support for **deprecated** environment variable?

Reply:
Why it's marked with `(Deprecated)`?

Reply:
So that people wouldn't write *new* code relying on usage of this.

Reply:
I shouldn't have asked this question in the first place as the answer seems to be obvious, but I'm surprised that you answered "No".

From the PR description:
> The old behavior could be temporarily restored via SYCL_PI_LEVEL_ZERO_EXPOSE_CSLICE_IN_AFFINITY_PARTITIONING environment variable but it is immediately deprecated, and customers are encouraged to switch to the new partitioning scheme as soon as possible.

From the sycl/doc/EnvironmentVariables.md:
> This option is introduced for compatibility reasons and is immediately deprecated.

It's very unfortunate that we have to add support for deprecated environment variables, but I don't see a clear option to keep the compatibility. Let's close this thread.